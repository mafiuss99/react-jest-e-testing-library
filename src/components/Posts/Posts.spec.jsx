import { render, screen } from "@testing-library/react";

import { Posts } from ".";

const props = {
    posts: [
        {
            id: 1,
            title: 'Title 1',
            body: 'body',
            cover: 'img/img1.png'
        },
        {
            id: 2,
            title: 'Title 2',
            body: 'body',
            cover: 'img/img1.png'
        },
        {
            id: 3,
            title: 'Title 3',
            body: 'body',
            cover: 'img/img1.png'
        }
    ]
}

describe('<Posts />', () => {
    it('should render posts', () => {
        render(<Posts {...props} />);

        expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(3);
        expect(screen.getAllByRole('img', { name: /title/i })).toHaveLength(3);
        expect(screen.getAllByText(/body/i)).toHaveLength(3);
    });

    it('should match snapshot', () => {
        const {container} = render(<Posts {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    })
});