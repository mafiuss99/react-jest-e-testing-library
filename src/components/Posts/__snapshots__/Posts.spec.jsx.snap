// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="Title 1"
      src="img/img1.png"
    />
    <div
      class="post-content"
    >
      <h2>
        Title 1
         
        1
      </h2>
      <p>
        body
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title 2"
      src="img/img1.png"
    />
    <div
      class="post-content"
    >
      <h2>
        Title 2
         
        2
      </h2>
      <p>
        body
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title 3"
      src="img/img1.png"
    />
    <div
      class="post-content"
    >
      <h2>
        Title 3
         
        3
      </h2>
      <p>
        body
      </p>
    </div>
  </div>
</div>
`;
